spec/classes/init_spec.rb

```diff
@@ -598,7 +598,80 @@
       }
 
       it { should_not contain_file('pam_system_auth_ac').with_content(/auth[\s]+sufficient[\s]+pam_vas3.so/) }
+      it {
+        should contain_file('pam_password_auth_ac').with({
+          'ensure'  => 'file',
+          'path'    => '/etc/pam.d/password-auth-ac',
+          'owner'   => 'root',
+          'group'   => 'root',
+          'mode'    => '0644',
+        })
+      }
+      it { should contain_file('pam_password_auth_ac').with_content("# This file is being maintained by Puppet.
+# DO NOT EDIT
+# Auth
+auth        required      pam_env.so
+auth        sufficient    pam_unix.so nullok try_first_pass
+auth        requisite     pam_succeed_if.so uid >= 500 quiet
+auth        required      pam_deny.so
+
+# Account
+account     required      pam_unix.so
+account     sufficient    pam_localuser.so
+account     sufficient    pam_succeed_if.so uid < 500 quiet
+account     required      pam_permit.so
+
+# Password
+password    requisite     pam_cracklib.so try_first_pass retry=3 type=
+password    sufficient    pam_unix.so sha512 shadow nullok try_first_pass use_authtok
+password    required      pam_deny.so
+
+# Session
+session     optional      pam_keyinit.so revoke
+session     required      pam_limits.so
+session     [success=1 default=ignore] pam_succeed_if.so service in crond quiet use_uid
+session     required      pam_unix.so
+")
+      }
+      it {
+        should contain_file('pam_password_auth').with({
+          'ensure' => 'symlink',
+          'path'   => '/etc/pam.d/password-auth',
+          'owner'  => 'root',
+          'group'  => 'root',
+        })
+      }
+    end
+
+  context 'pam_password_auth_ac set to a non absolute path on osfamily RedHat with operatingsystemmajrelease 6' do
+    let(:params) { { :pam_password_auth_ac => 'invalid/path' } }
+    let(:facts) do
+      { :osfamily          => 'RedHat',
+        :operatingsystemmajrelease => '6',
+      }
+    end
+
+    it 'should fail' do
+      expect {
+        should contain_class('pam')
+      }.to raise_error(Puppet::Error)
```

Comment:
Error's should have a message such as

```ruby
}.to raise_error(Puppet::Error/"invalid\/path" is not an absolute path/)
```

You can find these by putting in a failing spec test and then looking at the output from the tests.
```ruby
}.to raise_error(Puppet::Error/asdf/)
```